plugins/org.komodo.shell-api/src/org/komodo/shell/api/KomodoObjectLabelProvider.java

```diff
@@ -148,5 +152,19 @@
      *        the workspace status (never <code>null</code>)
      */
     void setWorkspaceStatus( final WorkspaceStatus status );
+    
+	 /**
+     * @return Returns a node names which should be omitted from display path
+     */
+	List<String> getGroupingNodes();
+	
+    /**
+     * Get the type display string for a KomodoObject
+     * @param uow the transaction
+     * @param kObj the KomodoObject
+     * @return the type display string
+     * @throws KException the exception
+     */
+    public String getTypeDisplay ( final Repository.UnitOfWork uow, final KomodoObject kObj );
```

Comment:
Can you add documentation for the pre- and post-conditions? For instance, identify if the args can be `null`and if the `return` can be empty. Also, `KException` is in the javadoc but not in the method signature. I would think *all* the implementers, except the default one, should return `null` if they cannot handle the specified `KomodoObject`.